coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -61,13 +69,46 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, fmt.Errorf("check prover task parameter failed, error:%w", err)
 	}
 
+	var hardForkNumber uint64
+	if getTaskParameter.HardForkName != "" {
+		var exist bool
+		hardForkNumber, exist = bp.nameForkMap[getTaskParameter.HardForkName]
+		if !exist {
+			log.Debug("hard fork prover get empty batch because of the hard fork name don't exist", "height",
+				getTaskParameter.ProverHeight, "fork name", getTaskParameter.HardForkName)
+			return nil, nil
```

Comment:
Why not return error?

Reply:
make sense

Reply:
fixed [32dc659](https://github.com/scroll-tech/scroll/pull/1182/commits/32dc6591862afe599ed881c73ad01156294e718e)